railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -621,7 +621,8 @@ def initialize(argv = ARGV)
       end
 
       def prepare!
-        handle_version_request!(@argv.first)
+        return handle_version_request!(@argv.first) if ["--version", "-v"].include?(@argv.first)
+
```

Comment:
Was this change necessary?

Reply:
[Test](https://github.com/rails/rails/pull/54365/files/ea0314dc76971fe3d391eedb41422602aebb372e#diff-a7343518cf4a5cac5405feeef22be122be1bce9359cc900bdab90c1122adc073R20) for version fails.

Reply:
I see, do we still need `exit(0)` then?

FYI, maybe interesting for you but it might help to use this from your rails checkout root when testing:

```
bundle exec railties/exe/rails -v new zomg
```

Reply:
👍 I will check it.

Reply:
Ok, so exit(0) is important and should stay as it is. I shouldn't modify the exit status. I can change a test because of the results of both behaviours are the same. Before changing and putting 'return' as a guard give me the same result as an output. Only test seams fail.